contao/templates/dcbe_general_treeview.html5

```diff
@@ -9,17 +9,40 @@ $GLOBALS['TL_CSS'][] = 'system/modules/dc-general/html/css/generalTreeView.css';
 
 <?php echo $this->getMessages(); ?>
 
-<div class="tl_listing_container tree_view table_<?php echo $this->tableName; ?>" id="tl_listing">
-	<?php if(strlen($this->breadcrumb)) echo $this->breadcrumb; ?>
+<?php if ($this->select): ?>
+<form action="<?php echo $this->action; ?>" id="tl_select" class="tl_form" method="post">
+    <div class="tl_formbody">
+        <input type="hidden" name="FORM_SUBMIT" value="tl_select">
+        <input type="hidden" name="REQUEST_TOKEN" value="<?php echo REQUEST_TOKEN; ?>">
+    <?php endif; ?>
+    <div class="tl_listing_container tree_view table_<?php echo $this->tableName; ?>" id="tl_listing">
+        <?php if(strlen($this->breadcrumb)) echo $this->breadcrumb; ?>
 
-    <ul class="tl_listing <?php echo $this->treeClass; ?>">
-        <li class="tl_folder_top">
-            <div class="tl_left">
-                <?php echo $this->strLabelIcon ?> <label><?php echo $this->strLabelText; ?></label>
+        <?php if ($this->select): ?>
+            <div class="tl_select_trigger">
+                <label for="tl_select_trigger" class="tl_select_label"><?php echo $GLOBALS['TL_LANG']['MSC']['selectAll']; ?></label>
```

Comment:
Changed it as proposed. I only touched this template as I didn't want to bloat this PR with other stuff.